### ISSUE TITLE
Cannot overwrite main template or partials

### ISSUE THREAD

**Issue by gruenebe** [nknapp/bootprint#9]
First of all I love the tool and you have it very well documented :)

But I am struggling to overwrite the main template. It will always render the default template and will not use my partials.

```
var path = require('path');

var bootprint = require('bootprint')
	.load(require('bootprint-swagger'))
	.merge({
		handlebars: {
			partials: path.resolve('./stuff/3rdparty/bootprint/partials'),
			template: path.resolve('./stuff/3rdparty/bootprint/template.hbs')
		}
	})
	.build('http://www.foo.bar/swagger.json','target')
	.generate()
	.done(console.log);
```

First of all, there is no message shown if the template path leads to a file which does not exist.
And even when it exists, it seems to be ignored.

I have *./stuff/3rdparty/bootprint/partials*

```
swagger/
    summary.hbs
    ...
```

*./stuff/3rdparty/bootprint/template.hbs*
```
{{!--
    @public
--}}
{{>base/header}}
<div class="container">
    {{>base/body}}
</div>
{{>base/footer}}

```

Or maybe I am just doing it wrong. Maybe you can help me with that :)

thank you.

**Comment by gruenebe** [comment]
Also cannot call `bootprint.generateHtml()`. It says undefined function.
API Doc says it can be used: https://github.com/nknapp/bootprint/blob/master/doc/api.md#bootprintgeneratehtml

**Comment by nknapp** [comment]
Hi and thanks for your report: I fear the project is very well documented, but parts of the documentation are a bit outdated since the last couple of weeks. I have changed some internals, but I forgot to build in backward compatibility code. Sorry for that.

* Try to replace `template: path.resolve('./stuff/3rdparty/bootprint/template.hbs')` by `templates: path.resolve('./stuff/3rdparty/bootprint/templates')`
* Template create the directory `./stuff/3rdparty/bootprint/templates` and put a file `index.html.hbs` in there.

The `generateHtml()`-function does not exist anymore. For developing your templates, try the `bootprint -d` option to run the file watcher. If you just want to use Handlebars without Less, you can use [customize](https://github.com/nknapp/customize), [customize-engine-handlebars](https://github.com/nknapp/customize-engine-handlebars) and [customize-write-files](https://github.com/nknapp/customize-write-files) to put together your own module.

**Comment by gruenebe** [comment]
Hi nknapp, thx for the update :)

I will try that soon. 

My use case is, that I want to inject the generated swagger html into a existing html page during my static-page build. Therefore I don't need the surrounding HTML code just the snippets inside `<body>`.
Therefore I want to use my own template.

And I want no less transformation since it slows my build down and I will provide my own static CSS.
What also bugged me was, that I cannot prevent that the bootstrap 3 CSS is included in the css.
But I will try the customize things you wrote about to disable it.

Right now I am already using bootstrap 4 alpha which replaced `panels` by `cards` and uses SCSS instead of LESS.
When I got some working bootstrap 4 templates for swagger maybe you are interested?

**Comment by nknapp** [comment]
Hi @gruenebe, 

sorry for the long time it took to answer this. I wasn't sure what to answer and then I have a lot of other issues in the past weeks (not all of them on github...)

I found this difficult to answer, because I'm a little torn apart about this:

On the one hand, Bootstrap 4 support would definitely be great and should be included without question.
On the other hand, there are definitely some people out there that have made customizations to their `bootprint-swagger` module and they may face compatibility issue if we just replace LESS by SCSS and change the Handlebars-templates to use Bootstrap 4 classes.
Although this is technically valid, because there is not 1.0 version yet, I don't like the idea of causing people problems, so my plan is this:

* I'm currently working on updating the documentation of the `bootprint-module`.
* Once that is done, I am going to release `1.0.0`-versions of `customize`, `customize-engine-less`, `customize-engine-handlebars`, `customize-watch`, `bootprint`, `bootprint-base`, `bootprint-json-schema` and `bootprint-swagger`.
* Then, we can immediately create a `2.x.x`-branch of those modules (as needed) and go for Bootstrap 4 compatibility in this branch. We also need a new module `customize-engine-scss` to do this.

If you are interested, I will give you push-access to the repositories, so that you don't have to create pull-requests.

What do you think of this?

**Comment by gruenebe** [comment]
Hi nknapp,

Your plan sounds good. But right now I am in the middle of pre release of my own software 
and was just looking for some tool to use. I switched back to Swagger UI for the moment.

Maybe in some months I will have time to get back to this topic, and maybe open another Pull Request. Since bootstrap4 is changing a lot lately it would be best to wait for a first stable release.

**Comment by nknapp** [comment]
That's great. Gives me some time too. I'm closing this until then. Good luck with your project...
